### PR TITLE
[WIP] Proj automatic download

### PR DESCRIPTION
**Pull request by SamuilsRulovs**
Implement some features as requested in the #226 

Created automatic download and configuration for:
- PROJ
- SQLite3
- libtiff
- CURL

### CODE REVIEW

**Comment by pjanck** [comment]
I'm missing content related to CURL?

File: CMakeLists.txt [hunk 1]
```diff
@@ -249,6 +249,22 @@ if(NOT blue_framework_POPULATED)
...
+
+FetchContent_Declare(
+		PROJ
+		GIT_REPOSITORY https://github.com/OSGeo/PROJ.git
+		GIT_TAG 7.1.1
+)
+
+FetchContent_GetProperties(PROJ)
+find_path(SQLITE3_INCLUDE_DIR NAMES sqlite3.h HINTS "C:/thirdparty/sqlite3")
+find_path(SQLITE3_LIBRARY NAMES sqlite3.dll HINTS "C:/thirdparty/sqlite3")
```
Comment (hechth) [inline CMakeLists.txt]: Use Find(SQLite3) as described here.

https://cmake.org/cmake/help/v3.19/module/FindSQLite3.html?highlight=sqlite

**Comment by pjanck** [comment]
And, as discussed together, use the `OSGeo4W` [installer](https://trac.osgeo.org/osgeo4w/) to install `proj4`.

**Comment by pjanck** [comment]
See #252 .

**Pull request closed without merging**